libraries/app/application.cpp

```diff
@@ -397,12 +398,21 @@ void application_impl::startup()
    if( _options->count("replay-blockchain") || _options->count("revalidate-blockchain") )
       _chain_db->wipe( _data_dir / "blockchain", false );
 
-   if( _options->count("revalidate-blockchain") )
-      _chain_db->_replay_with_validation = true;
-
    try
    {
-      _chain_db->open( _data_dir / "blockchain", initial_state, GRAPHENE_CURRENT_DB_VERSION );
+      // these flags are used in open() only, i. e. during replay
+      uint32_t skip = graphene::chain::database::skip_witness_signature |
+                      graphene::chain::database::skip_block_size_check |
+                      graphene::chain::database::skip_merkle_check |
+                      graphene::chain::database::skip_transaction_signatures |
+                      graphene::chain::database::skip_transaction_dupe_check |
+                      graphene::chain::database::skip_tapos_check |
+                      graphene::chain::database::skip_witness_schedule_check;
+      if( _options->count("revalidate-blockchain") ) // see also handle_block()
+         skip = _options->count("force-validate") ? 0 : graphene::chain::database::skip_transaction_signatures;
```

Comment:
1. I think it's better to replace `0` with `graphene::chain::database::skip_nothing`.
2. Code readability here looks a bit poor. Now it reads:
```
declare skip = something;
if (condition A)
{
   skip = condition B ? something else : yet something else;
}
```

How about change the `if` to `?:`
```
declare skip = condition A ? condition B ? something else : yet something else : something;
```
or change the `?:` to `if else`?

Reply:
done